dot/core/service.go

```diff
@@ -266,6 +269,9 @@ func (s *Service) handleCodeSubstitution(hash common.Hash,
 // does not need to be completed before the next block can be imported.
 func (s *Service) handleBlocksAsync() {
 	for {
+		// TODO Look into if this is best way to get prev
+		// Maybe pass in block and get parent rather than assume that
+		// best block hash is the parent
 		prev := s.blockState.BestBlockHash()
```

Comment:
Leaving this comment here to get reviewers thoughts on it

Reply:
Is previous meant to be the parent of the block? If so, we should definitely get the parent of the block by using `block.ParentHash` to look up its parent block right?

Reply:
I think I would agree. I will hold up on changing till I get more eyes on it

Reply:
I think this is right.  We are trying to compare the best block to the incoming block we receive over the channel.  The channel is produced by `handleBlock()` via `HandleBlockImport` and `HandleBlockProduced`.  So we may receive blocks from other authors that may have forked, and that's what we're trying to detect when we call `handleChainReorg` when we receive a block over the channel.

I think what's wrong is that we do `prev := s.blockState.BestBlockHash()` but `prev` isn't used until line 288.  This method will block on like 278 until either the context is cancelled or a block comes in through `s.blockAddCh`.  So there's a probability we're using an outdated best block.

Reply:
@timwu20 I talked to Elizabeth and she brought up a good point, why do we not use the block returned in the channel? I feel like we should use that instead of getting best block. Thoughts?